aten/src/ATen/native/cuda/ScaledGroupMM.cu

```diff
@@ -521,9 +499,294 @@ void dispatch_fp8_grouped_gemm_on_bias_dtype(
+
+  using CollectiveEpilogue =
+      typename cutlass::epilogue::collective::CollectiveBuilder<
+          ArchTag, OperatorClass, typename Config::MmaTileShape,
+          typename Config::ClusterShape,
+          cutlass::epilogue::collective::EpilogueTileAuto, DtypeAccum,
+          DtypeAccum, DtypeOutput, LayoutOutput *, AlignmentOutput, DtypeOutput,
+          LayoutOutput *, AlignmentOutput,
+          typename Config::EpilogueSchedule>::CollectiveOp;
+
+  // Collective Mainloop
+  using CollectiveMainloop =
+      typename cutlass::gemm::collective::CollectiveBuilder<
+          ArchTag, OperatorClass, DtypeA,
+          cute::tuple<cutlass::layout::RowMajor *,
+                      typename Config::LayoutSFA *>,
+          AlignmentA, DtypeB,
+          cute::tuple<cutlass::layout::ColumnMajor *,
+                      typename Config::LayoutSFB *>,
+          AlignmentB, DtypeAccum, typename Config::MmaTileShape,
+          typename Config::ClusterShape,
+          cutlass::gemm::collective::StageCountAutoCarveout<static_cast<int>(
+              sizeof(typename CollectiveEpilogue::SharedStorage))>,
+          typename Config::KernelSchedule>::CollectiveOp;
+
+  using ProblemShape =
+      cutlass::gemm::GroupProblemShape<cute::Shape<int, int, int>>;
+
+  using GemmKernel =
+      cutlass::gemm::kernel::GemmUniversal<ProblemShape, CollectiveMainloop,
+                                           CollectiveEpilogue>;
+  using Gemm = cutlass::gemm::device::GemmUniversalAdapter<GemmKernel>;
+
+  using StrideA = typename Gemm::GemmKernel::InternalStrideA;
+  using StrideB = typename Gemm::GemmKernel::InternalStrideB;
+  using StrideOutput = typename Gemm::GemmKernel::InternalStrideD;
+
+  using DtypeProblemShape = cutlass::gemm::GroupProblemShape<
+      cute::Shape<int, int, int>>::UnderlyingProblemShape;
+
+  auto group_count_info = at::cuda::detail::get_group_count(mat_a, mat_b, offs);
+  auto [M, N, K, group_count, type] = group_count_info;
+
+  int aligned_group_count =
+      round_up_to_nearest_multiple(group_count, 16 / int(sizeof(void *)));
+
+  auto ptr_opts = at::TensorOptions().device(mat_a.device()).dtype(at::kLong);
+  auto stride_opts =
+      at::TensorOptions().device(mat_a.device()).dtype(at::kLong);
+  auto shape_opts = at::TensorOptions().device(mat_a.device()).dtype(at::kInt);
+
+  at::Tensor inputA_ptrs = at::empty({aligned_group_count}, ptr_opts);
+  at::Tensor inputB_ptrs = at::empty({aligned_group_count}, ptr_opts);
+  at::Tensor output_ptrs = at::empty({aligned_group_count}, ptr_opts);
+  at::Tensor inputA_scale_ptrs = at::empty({aligned_group_count}, ptr_opts);
+  at::Tensor inputB_scale_ptrs = at::empty({aligned_group_count}, ptr_opts);
+
+  at::Tensor stride_A = at::empty({group_count}, stride_opts);
+  at::Tensor stride_B = at::empty({group_count}, stride_opts);
+  at::Tensor stride_output = at::empty({group_count}, stride_opts);
+  at::Tensor problem_sizes = at::empty({group_count, 3}, shape_opts);
```

Comment:
The original SM90 implementation uses a contiguous allocation and manually computes the offsets for each tensor:
[https://github.com/pytorch/pytorch/blob/f9ba45b88c4885f2eb61c4db16a4293f7535a1f0/aten/src/ATen/native/cuda/ScaledGroupMM.cu#L311](https://github.com/pytorch/pytorch/blob/f9ba45b88c4885f2eb61c4db16a4293f7535a1f0/aten/src/ATen/native/cuda/ScaledGroupMM.cu#L311)
However, I found that approach a bit hard to read, so I opted to use `at::empty` instead. Let me know if there are any concerns with this approach.

Reply:
Yes, please revert to the old implementation, there's no reason to pay microsecond overheads on dozens of allocations